source/codegen/metadata_mutation.py

```diff
@@ -0,0 +1,51 @@
+import common_helpers
+
+RESERVED_WORDS = [
+    'abstract', 'as',
+    'base', 'bool', 'break','byte',
+    'case', 'catch', 'char', 'checked', 'class', 'const', 'continue',
+    'decimal', 'default', 'delegate', 'do', 'double',
+    'else', 'enum', 'event', 'explicit', 'extern',
+    'false', 'finally', 'fixed', 'float', 'for', 'foreach',
+    'goto',
+    'if', 'implicit', 'in', 'int', 'interface', 'internal', 'is',
+    'lock', 'long',
+    'namespace', 'new', 'null',
+    'object', 'operator', 'out', 'override',
+    'params', 'private', 'protected', 'public',
+    'readonly', 'ref', 'return',
+    'sbyte', 'sealed', 'short', 'sizeof', 'stackalloc', 'static', 'status', 'string', 'struct', 'switch',
+    'this', 'throw', 'true', 'try', 'typeof',
+    'uint', 'ulong', 'unchecked', 'unsafe', 'ushort', 'using',
+    'virtual', 'void', 'volatile',
+    'while'
+]
+
+def sanitize_names(parameters):
+    """Sanitizes name fields on a list of parameter objects and populates the cppname field with the sanitized value."""
+    for parameter in parameters:
+        parameter['cppName'] = parameter['name']
+        if parameter['cppName'] in RESERVED_WORDS:
+            parameter['cppName'] += 'Parameter'
+
+def mark_size_params(parameters):
+    """Marks the size parameters in the metadata."""
+    for param in parameters:
+        mechanism = common_helpers.get_size_mechanism(param)
+        if mechanism in {'len', 'ivi-dance', 'passed-in'}:
+            size_param = get_size_param(param, parameters)
+            size_param['is_size_param'] = True
+
+def mark_non_grpc_params(parameters):
```

Comment:
I think `non_proto_params` would be better since we're removing it from the .proto file but still referring to it in the service handler.

Reply:
Agreed, that's more accurate. Updated the name and description taking the `non_proto_params` terminology into account.